org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -441,9 +454,70 @@ public static IJavaElement findElementAtSelection(ITypeRoot unit, int line, int
 		if (offset > -1) {
 			return unit.codeSelect(offset, 0);
 		}
+		if (unit instanceof IClassFile) {
+			IClassFile classFile = (IClassFile) unit;
+			String contents = ClassfileContentHandler.disassemble(classFile);

```

Comment:
Can we move the disassembling method to a utility method, I'd rather prefer if JDTUtil didn't depend on a handler